cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -652,38 +625,109 @@ def to_schema(self, rules: DMSRules) -> DMSSchema:
-            )
-        if self.instance_space:
-            data_model.space = self.instance_space
-            spaces.append(dm.SpaceApply(space=self.instance_space, name=self.instance_space))
+        node_types = dm.NodeApplyList([])
+        parent_views = {parent for view in views for parent in view.implements or []}
+        node_type_flag = False
+        for view in views:
+            ref_containers = view.referenced_containers()
+            has_data = dm.filters.HasData(containers=list(ref_containers)) if ref_containers else None
+            node_type = dm.filters.Equals(["node", "type"], {"space": view.space, "externalId": view.external_id})
+            if view.as_id() in parent_views:
+                view.filter = has_data
+            elif has_data is None:
+                # Child filter without container properties
+                if node_type_flag:
+                    # Transformations do not yet support setting node type.
+                    view.filter = node_type
+                    node_types.append(dm.NodeApply(space=view.space, external_id=view.external_id, sources=[]))
+            else:
+                # Child filter with its own container properties
+                if node_type_flag:
+                    # Transformations do not yet support setting node type.
+                    view.filter = dm.filters.And(has_data, node_type)
+                    node_types.append(dm.NodeApply(space=view.space, external_id=view.external_id, sources=[]))
+                else:
+                    view.filter = has_data
+        return views, node_types
+
+    def _create_containers(
+        self,
+        dms_container: SheetList[DMSContainer] | None,
+        container_properties_by_id: dict[dm.ContainerId, list[DMSProperty]],
+        default_space: str,
+    ) -> dm.ContainerApplyList:
+        containers = dm.ContainerApplyList(
+            [
+                dms_container.as_container(default_space, self.standardize_casing)
+                for dms_container in dms_container or []
+            ]
+        )
+        container_to_drop = set()
+        for container in containers:
+            container_id = container.as_id()
+            if not (container_properties := container_properties_by_id.get(container_id)):
+                warnings.warn(issues.dms.EmptyContainerWarning(container_id=container_id), stacklevel=2)
+                container_to_drop.add(container_id)
+                continue
+            for prop in container_properties:
+                if prop.container_property is None:
+                    continue
+                if isinstance(prop.value_type, DMSValueType):
+                    type_cls = prop.value_type.dms
+                else:
+                    type_cls = dm.DirectRelation
+
+                prop_name = to_camel(prop.container_property) if self.standardize_casing else prop.container_property
```

Comment:
For time being this is fine, but we already have under SheetEntity defined option to add `name` field, in case for example when property_ and class_ (i.e. ids) are human unreadable, which is the case for one of our clients.

Reply:
Not sure if I understand this. Are you suggesting a better way to handle properties?